test/vm_tests/builder/declaration_parsing_test.dart

```diff
@@ -1538,7 +1538,8 @@ main() {
                 ]));
                 final decl = declarations.firstWhereType<PropsMapViewOrFunctionComponentDeclaration>();
 
-                expect(decl.factory?.name?.name, name);
+                expect(decl.factories.length, 1);
```

Comment:
Prefer using `hasLength`, which gives better failure messages
```suggestion
                expect(decl.factories, hasLength(1));
```